src/mps/mps.jl

```diff
@@ -583,11 +584,48 @@ Czz = correlation_matrix(psi,"Sz","Sz")
 s = siteinds("Electron",N; conserve_qns=true)
 psi = randomMPS(s, n->isodd(n) ? "Up" : "Dn"; linkdims=m)
 Cuu = correlation_matrix(psi,"Cdagup","Cup";site_range=2:8)
-```
+``` 
 """
-function correlation_matrix(psi::MPS, Op1::AbstractString, Op2::AbstractString; kwargs...)
+function correlation_matrix(psi::MPS, _Op1::AbstractString, _Op2::AbstractString; kwargs...)
   N = length(psi)
   ElT = promote_itensor_eltype(psi)
+  s = siteinds(psi)
+
+  Op1 = _Op1 #make copies into which we can insert "F" string operators, and then restore.
+  Op2 = _Op2
+  onsiteOp = "$Op1*$Op2"
+  fermionic1 = has_fermion_string(Op1, s[1])
+  fermionic2 = has_fermion_string(Op2, s[1])
+  if fermionic1 != fermionic2
+    error(
+      "correlation_matrix: Mixed fermionic and bosonic operators are not supported yet."
+    )
+  end
+
+  # Decide if we need to calculate a non-hermitian corr. matrix which is roughly double the work.
+  is_cm_hermitian = false #Assume corr-matrix is non-hermitian
+  if haskey(kwargs, :is_hermitian) #Did the user explicitly request something?
```

Comment:
Sorry to nitpick, but I would prefer the name `ishermitian` since that is the name of the Julia function:
```julia
julia> using LinearAlgebra

help?> ishermitian
search: ishermitian

  ishermitian(A) -> Bool


  Test whether a matrix is Hermitian.

[...]
```
and we use `ishermitian` as a keyword argument in other functions, like `eigen`.